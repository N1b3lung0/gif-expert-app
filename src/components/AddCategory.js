import React, { useState } from 'react'
import PropTypes from 'prop-types';
import { InputText } from 'primereact/inputtext';


export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => setInputValue( e.target.value );

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setCategories( categories => [inputValue, ...categories] );
            setInputValue('');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <span className="p-float-label">
                <InputText
                    id="float-input"
                    type="text"
                    value={ inputValue }
                    onChange={ handleInputChange }
                />
                <label htmlFor="float-input">Search</label>
            </span>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}