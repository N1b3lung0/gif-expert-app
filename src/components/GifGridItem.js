import 'primereact/resources/themes/nova-dark/theme.css';
import 'primereact/resources/primereact.css';

import React from 'react'
import PropTypes from 'prop-types';
import { Card } from 'primereact/card';

export const GifGridItem = ({ title, url}) => {
    
    const header = ( <img src={ url } alt={ title }/> );

    return (
        <Card
            subTitle={ title }
            style={{width: '360px'}}
            className="ui-card-shadow card animate__animated animate__fadeIn"
            header={ header }
        />
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}

Card.propTypes = {
    subTitle: PropTypes.string.isRequired,
    header: PropTypes.object.isRequired
}
