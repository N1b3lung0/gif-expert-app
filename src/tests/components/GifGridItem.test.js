import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';
import { Card } from 'primereact/card';

describe('Testing GifGridItem', () => {

    const title = 'titulo';
    const url = 'https://localhost/imagen.jpg';
    const header = ( <img src={ url } alt={ title }/> );
    
    test('should show the component GifGridItem correctly', () => {
        const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );
        expect(wrapper).toMatchSnapshot();
    });

    test('should show the component Card correctly', () => {
        const wrapper = shallow( <Card subTitle={ title } header={ header } /> );
        expect(wrapper).toMatchSnapshot();
    });
    
    
})
