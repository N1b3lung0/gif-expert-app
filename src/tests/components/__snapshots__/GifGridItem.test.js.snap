// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGridItem should show the component Card correctly 1`] = `
<div
  className="p-card p-component"
  style={null}
>
  <div
    className="p-card-header"
  >
    <img
      alt="titulo"
      src="https://localhost/imagen.jpg"
    />
  </div>
  <div
    className="p-card-body"
  >
    <div
      className="p-card-subtitle"
    >
      titulo
    </div>
  </div>
</div>
`;

exports[`Testing GifGridItem should show the component GifGridItem correctly 1`] = `
<Card
  className="ui-card-shadow card animate__animated animate__fadeIn"
  footer={null}
  header={
    <img
      alt="titulo"
      src="https://localhost/imagen.jpg"
    />
  }
  id={null}
  style={
    Object {
      "width": "360px",
    }
  }
  subTitle="titulo"
  title={null}
/>
`;
